packages/fireproof/tsup.config.ts

```diff
@@ -0,0 +1,114 @@
+import { defineConfig, Options } from 'tsup';
+import alias from "esbuild-plugin-alias"
+import path from "path"
+
+const LIBRARY_BUNDLE_OPTIONS: Options = {
+  format: ["esm", "cjs", "iife"],
+  target: ["esnext", "node18"],
+  globalName: "Fireproof",
+  clean: true,
+  sourcemap: true,
+  metafile: true,
+  minify: false,
+}
+
+const LIBRARY_BUNDLES: readonly Options[] = [
+  {
+    ...LIBRARY_BUNDLE_OPTIONS,
+    name: "core/browser",
+    entry: ["src/fireproof.ts"],
+    platform: "browser",
+    outDir: "dist/browser",
+    dts: {
+      footer: "declare module '@fireproof/core'"
+    },
+  },
+  {
+    ...LIBRARY_BUNDLE_OPTIONS,
+    name: "core/node",
+    entry: ["src/fireproof.ts"],
+    platform: "node",
+    outDir: "dist/node",
+    dts: {
+      footer: "declare module '@fireproof/core'"
+    },
+  },
+  {
+    ...LIBRARY_BUNDLE_OPTIONS,
+    name: "core/memory",
+    entry: ["src/fireproof.ts"],
+    platform: "browser",
+    outDir: "dist/memory",
+    dts: {
+      footer: "declare module '@fireproof/core'"
+    },
+  },
```

Comment:
Aside from the tests not passing, I believe I'm doing this `memory` export wrong and could use direction on what it should truly be which I assume to be related to leveraging the `eb-edge` module perhaps?

Reply:
The `eb-edge.ts` file is kind of a placeholder today, I just took a guess at what makes sense in the test runtime platform. We should expect to change the file as we tune into the runtime requirements (depending on eg what storage is available on github actions, etc)